crates/nu-cli/tests/completions/mod.rs

```diff
@@ -390,6 +390,34 @@ fn external_commands_only() {
     match_suggestions(&expected, &suggestions);
 }
 
+/// Disable external commands except for those start with `^`
+#[test]
+fn external_commands_disabled() {
+    let mut engine = new_external_engine();
+
+    let mut config = Config::default();
+    config.completions.external.enable = false;
+    engine.set_config(config);
+
+    let stack = nu_protocol::engine::Stack::new();
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+    let completion_str = "ls; ^sleep";
+    let suggestions = completer.complete(completion_str, completion_str.len());
+    #[cfg(windows)]
+    let expected: Vec<_> = vec!["sleep.exe"];
+    #[cfg(not(windows))]
+    let expected: Vec<_> = vec!["sleep"];
+    match_suggestions(&expected, &suggestions);
+
+    let completion_str = "sleep";
+    let suggestions = completer.complete(completion_str, completion_str.len());
+    #[cfg(windows)]
+    let expected: Vec<_> = vec!["sleep"];
+    #[cfg(not(windows))]
```

Comment:
Doesn't seem like we need a separate case for Windows here
```suggestion
```

Reply:
Nice catch!